[flay test] test_rawkv::test_raw_put_key_guard

## Bug Report

<!-- Thanks for your bug report! Don't worry if you can't fill out all the sections. -->

### What version of TiKV are you using?
<!-- You can run `tikv-server --version` -->

master (b1715ea3070d1d0dc8a81e1cd8b5ce13b8129ee3)

### What operating system and CPU are you using?
<!-- If you're using Linux, you can run `cat /proc/cpuinfo` -->

AMD64

### Steps to reproduce
<!-- If possible, provide a recipe for reproducing the error. A complete runnable program is good. -->

```
❯ ./scripts/env cargo test -p tests --test failpoints test_rawkv -- --nocapture
```

### What did you expect?

Test succeed.

### What did happened?

Test failed:

```
❯ ./scripts/env cargo test -p tests --test failpoints test_rawkv -- --nocapture
    Finished test [unoptimized] target(s) in 0.65s
warning: the following packages contain code that will be rejected by a future version of Rust: criterion v0.3.5, nom v2.2.1, nom v4.2.3, nom v5.1.0, quick-xml v0.22.0
note: to see what the problems were, use the option `--future-incompat-report`, or run `cargo report future-incompatibilities --id 3`
     Running failpoints/mod.rs (target/debug/deps/failpoints-04222e95374eb0c8)

running 3 tests
test cases::test_rawkv::test_leader_transfer ... ok
thread 'cases::test_rawkv::test_raw_put_key_guard' panicked at tests/failpoints/cases/test_rawkv.rs:318:23:
called `Option::unwrap()` on a `None` value
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
test cases::test_rawkv::test_raw_put_key_guard ... FAILED
test cases::test_rawkv::test_region_merge ... ok

failures:

failures:
    cases::test_rawkv::test_raw_put_key_guard

test result: FAILED. 2 passed; 1 failed; 0 ignored; 0 measured; 365 filtered out; finished in 9.00s

error: test failed, to rerun pass `-p tests --test failpoints`
make: *** [scripts/../Makefile:317: run] Error 101
❯
```

--- a/tests/failpoints/cases/test_rawkv.rs
+++ b/tests/failpoints/cases/test_rawkv.rs
@@ -290,7 +290,8 @@ fn test_raw_put_key_guard() {
     let region_id = region.get_id();
     let client = suite.get_client(region_id);
     let ctx = suite.get_context(region_id);
-    let node_id = region.get_peers()[0].get_id();
+    let leader = suite.cluster.leader_of_region(region_id).unwrap();
+    let node_id = leader.get_id();
     let leader_cm = suite.cluster.sim.rl().get_concurrency_manager(node_id);
     let ts_provider = suite.get_causal_ts_provider(node_id).unwrap();
     let ts = block_on(ts_provider.async_get_ts()).unwrap();
@@ -305,9 +306,10 @@ fn test_raw_put_key_guard() {
     // Wait for global_min_lock_ts.
     sleep_ms(500);
     let start = Instant::now();
-    while leader_cm.global_min_lock_ts().is_none()
-        && start.saturating_elapsed() < Duration::from_secs(5)
-    {
+    while leader_cm.global_min_lock_ts().is_none() {
+        if start.saturating_elapsed() > Duration::from_secs(5) {
+            panic!("wait for global_min_lock_ts timeout");
+        }
         sleep_ms(200);
     }
 

